input/common/execread/execread.go

```diff
@@ -66,60 +79,83 @@ func (s *Session) Start(ctx context.Context, dst [][]input.Sample, kickChan chan
 		f64:   !s.f32mode,
 	}
 
-	// Allocate 4 times the buffer. We should ensure that we can read some of
-	// the overflow.
-	raw := make([]byte, bufsz)
-
-	if err := cmd.Start(); err != nil {
-		return errors.Wrap(err, "failed to start ffmpeg")
+	bufsz := s.samples
+	if !s.f32mode {
+		bufsz *= 2
 	}
 
-	for {
-		reader.reset(raw)
+	raw := make([]byte, bufsz*4)
 
-		mu.Lock()
-		for n := 0; n < s.samples; n++ {
-			dst[n%framesz][n/framesz] = reader.next()
-		}
-		mu.Unlock()
+	sampleDuration := time.Duration(
```

Comment:
currently the way the buffer is read means this will sometimes be too short. can we double this for now and fix it properly later with some refactoring?

Reply:
Yeah, I made it so that it's 2x until the read expires once, then it's 1x for all future deadlines until it's sending samples again.